nuxeo-compound-documents-web/test/nuxeo-compound-document-tree.test.js

```diff
@@ -0,0 +1,826 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import {
+  fixture,
+  flush,
+  html,
+  isElementVisible,
+  login,
+  tap,
+  waitForAttrMutation,
+  waitForChildListMutation,
+} from '@nuxeo/testing-helpers';
+import { expect } from '@esm-bundle/chai';
+import '@nuxeo/nuxeo-ui-elements/nuxeo-icons.js';
+import '@polymer/iron-icons/iron-icons.js';
+import '@polymer/iron-icons/hardware-icons.js';
+import '../elements/nuxeo-compound-document-tree.js';
+
+let rootDocument;
+let levelOneDocuments;
+let levelTwoDocuments;
+let levelThreeDocuments;
+let levelOneDocument;
+let levelTwoDocument;
+let uidCounter;
+
+// mock the label for the document tree root
+window.nuxeo.I18n.language = 'en';
+window.nuxeo.I18n.en = window.nuxeo.I18n.en || {};
+window.nuxeo.I18n.en['browse.root'] = 'Root';
+
+const jsonHeader = { 'Content-Type': 'application/json' };
+
+// generate a document for the tree
+const generateDocument = ({ type, parentRef, parentPath, isFolderish, isCompound, hasThumbnail }) => {
+  const uid = uidCounter;
+  uidCounter += 1;
+  const doc = {
+    'entity-type': 'document',
+    repository: 'default',
+    uid: `${uid}`,
+    path: type === 'Root' ? '/' : `/${parentPath}/${type}${uid}`,
+    type,
+    parentRef,
+    title: type === 'Root' ? type : `${type}${uid}`,
+    properties: {},
+    facets: ['HiddenInCreation', 'NotCollectionMember'],
+    schemas: [
+      {
+        name: 'common',
+        prefix: 'common',
+      },
+      {
+        name: 'dublincore',
+        prefix: 'dc',
+      },
+    ],
+    contextParameters: {
+      hasFolderishChild: isFolderish,
+      hasContent: isCompound,
+    },
+  };
+  if (isFolderish) {
+    doc.facets.push('Folderish');
+  }
+  if (isCompound) {
+    doc.facets.push('CompoundDocument');
+  }
+  if (hasThumbnail) {
+    doc.contextParameters.thumbnail = {
+      url: '/test/images/icon.png',
+    };
+  }
+  return doc;
+};
+
+// generate the page provider response
+const generatePageProviderResponse = (entries = []) => {
+  return {
+    'entity-type': 'documents',
+    isPaginable: true,
+    resultsCount: entries.length,
+    pageSize: 40,
+    maxPageSize: 40,
+    resultsCountLimit: 40,
+    currentPageSize: entries.length,
+    currentPageIndex: 0,
+    currentPageOffset: 0,
+    numberOfPages: 1,
+    isPreviousPageAvailable: false,
+    isNextPageAvailable: false,
+    isLastPageAvailable: false,
+    isSortable: true,
+    totalSize: entries.length,
+    pageIndex: 0,
+    pageCount: 1,
+    entries,
+  };
+};
+
+/**
+ * Mock router.
+ * calling a global javascript function to prevent URL redirect.
+ */
+const router = {
+  browse: (path) => path.substring(1),
+  document: (path) => path,
+};
+
+const getTreeRoot = (el) => el.$$('nuxeo-tree-node#root');
```

Comment:
nit: prefer standard APIs
```suggestion
const getTreeRoot = (el) => el..shadowRoot.querySelector('nuxeo-tree-node#root');
```